moto/sagemaker/utils.py

```diff
@@ -0,0 +1,16 @@
+import boto3
+import json
+
+
+def load_pipeline_definition_from_s3(pipeline_definition_s3_location):
+    client = boto3.client("s3")
```

Comment:
It will be quicker to use the internal API to retrieve data from other services:
```
from moto.s3.models import s3_backends
s3_backend = s3_backends[account_id]["global"]
s3_backend.get_object(...)
```

Reply:
Thanks!

Reply:
I've been testing this implementation variant. However, I keep running into retrieving a `FakeKey` object from `get_object` then, which I cannot retrieve the original content from (at least I haven't found out how yet). Therefore, I can't really mimick the original chain of events. I'm wondering what I am not seeing or doing right here when it comes to handling the retrieved FakeKey via `get_object`?

https://github.com/spulec/moto/actions/runs/3756496788/jobs/6382939745

Reply:
You can use the value-attribute: `json.loads(key.value)`

https://github.com/spulec/moto/blob/master/moto/s3/models.py#L162

We don't have type-annotations yet for S3, but this property should return the original content as `bytes`

Reply:
Thanks, now I got it working!

Reply:
Unfortunately, I can't seem to get the Server Mode tests working as the mocked S3 buckets can't be found... Do you perhaps have pointer here?

Reply:
Hi @stiebels, apologies for the delay. In ServerMode, Moto runs inside a separate Python runtime, making it impossible to reach the S3 data directly. You would have to use boto3 to download the pipeline definition (`boto3.client("s3").download_object(..)`)

Reply:
Thanks! Hope you've had a nice christmas break!